### PR TITLE
[DNM] Adds plasma plasmifying exposed limbs

### PR DESCRIPTION
**Pull request by ghost**
Exposing your bare skin to plasma and getting some extreme plasma toxins may lead to mutation of the limb.

### CODE REVIEW

File: code/ZAS/NewSettings.dm [hunk 1]
```diff
@@ -222,6 +222,17 @@ var/global/ZAS_Settings/zas_settings = new
...
 
+/datum/ZAS_Setting/PLASMA_MUTATION
+	name = "Plasmaman Mutation"
+	desc = "Does being in plasma cause a persons exposed limbs to turn into that of a plasmaman?"
+	value = 1
+	valtype=ZAS_TYPE_BOOLEAN
+
+/datum/ZAS_Setting/PLASMA_MUTATION_CHANCE
+	name = "Plasmaman mutation chance divider"
+	desc = "Divides the chance for an exposed limb to plasmify. 5 = 1/5th of toxin damage chance."
```
Comment (ComicIronic) [inline code/ZAS/NewSettings.dm]: Something about this number description is a bit weird to me. I get what you're explaining, but maybe you could phrase it better?

Comment (ghost) [reply code/ZAS/NewSettings.dm]: maybe I should have it as a multiplier instead and it start at 0.2?

File: code/modules/organs/organ_external.dm [hunk 2]
```diff
@@ -961,6 +961,18 @@ Note that amputating the affected organ does in fact remove the infection from t
 	species = new S.type
 	owner.update_body()
 
+/datum/organ/external/proc/plasmify()
+	if(istype(species, /datum/species/plasmaman))
```
Comment (ComicIronic) [inline code/modules/organs/organ_external.dm:965]: Use the defines for these, the same way you used `isvox`.

**Review by ComicIronic — DISMISSED** [review]

File: code/ZAS/NewSettings.dm (hunk 1 above)
Comment (ComicIronic) [reply code/ZAS/NewSettings.dm]: Yeah, that would make more sense. Ideally, can you write out the equation in the description?

File: code/modules/organs/organ_external.dm (hunk 2 above)
Comment (ghost) [reply code/modules/organs/organ_external.dm:965]: the macros don't work in this case, I really shouldn't be using isvox()

isvox, or isplasmaman, checks if the current type is /mob/living/carbon/human, which would not be true for an external organ datum.

**Comment by Blithering** [comment]
Can this lead to people fully turning into plasmamen?

**Comment by ghost** [comment]
yes, if entirely exposed

**Comment by Blithering** [comment]
Neat, was just thinking this should be a thing the other day. Get out of my head!

**Comment by Kuranon** [comment]
how much plasma is extreme plasma

**Comment by carshalash** [comment]
Apply -28, -20 to the co ordinates of the telepad for a sawbones approved makeover.

**Comment by Kammerjunk** [comment]
@Kuranon 
`if(prob(getToxLoss())*zas_settings.Get(/datum/ZAS_Setting/PLASMA_MUTATION_MULTIPLIER))`
So every time it runs and you're exposed to plasma, that's a chance of (your toxin damage times 0.2) to increment your `plasma_limb_count`.

`if(victim && prob(100/plasma_limb_count))`
After 100 runs, that's a 1% chance per run to ghoulify.

I'm not calling them ticks here. Are they the same ticks as Life? I dunno.

**Comment by Shadowmech88** [comment]
If you were naked in a plasmaflood, what are the odds that you would get turned into a plasmaman and survive rather than dying from toxins?

**Comment by Exxion** [comment]
Didn't we decide not to do this like months ago

**Comment by ihadtoregisterforthis** [comment]
The other plasmamorphing PR made full plasmamen on death

**Comment by ghost** [comment]
@Kammerjunk not quite.

Each tick you're in plasma, there's a probability check of your current toxin damage times'd be the plasma_mutation_multiplier which is 0.2

Should it succeed, it runs through your limbs and checks to see if it is not amputated, organic, or plasmaman or covered. Should it find a plasmaman limb then plasma_limb_count is incremented. Should it find a possible limb to change, then it's a probability fire of 100/plasmaman limb count for the exposed limb to mutate.

**Comment by ihadtoregisterforthis** [comment]
What does this do to plasmamen? 
What does it do to greys?
What does it do to aliens?

**Comment by ghost** [comment]
plasmamen it does nothing, as their limbs'll pass through the plasmaman check and be exempt.

All playable species bar vox and plasmamen can get plasmified.

Xenos it's the same old business of them getting plasma-toxin from plasmafloods for rapid egg laying and such

**Comment by ghost** [comment]
Wouldn't this be bad since you ignite if there's oxygen in the Atmos and start a plasma fire if you turn into plasmameme? Sounds fun

**Comment by Shadowmech88** [comment]
Yeah, I didn't think about that. Wouldn't this allow any plasmaflood to spontaneously turn into a plasmafire if someone is caught in it unprotected?

**Comment by ghost** [comment]
That's a bit of an interesting question right now, as why would anyone be naked in a plasmaflood, and this does not change the persons base species into that of a plasmaman, just how the limbs look and their compatibility.

Bit of an oversight, now that I look at it.

**Comment by ReindH** [comment]
"why would anyone be naked in a plasmaflood"
Fresh clones, people freshly out of the healtubes.

**Comment by Kammerjunk** [comment]
Just shove them into cryo with their clothes on :^)

**Comment by Dacendeth** [comment]
Does this only work on fleshy limbs?

**Comment by ghost** [comment]
works on skeletal limbs too.

Ooh, good shout, could do with adding a check for skeletal vox as they're exempt too.

No chicken flambe around here

**Comment by Dacendeth** [comment]
So it would turn robotic limbs into plasmalimbs?

**Comment by ghost** [comment]
No, it checks for non-organics such as robotic or peg

**Comment by Blithering** [comment]
RIP

**Comment by ghost** [comment]
concept was flawed. Gonna reapproach in a different manner

**Pull request closed without merging**